include/shark/Algorithms/DirectSearch/Operators/Grid.h

```diff
@@ -0,0 +1,251 @@
+//===========================================================================
+/*!
+ * 
+ *
+ * \brief       Various functions for generating n-dimensional grids.
+ * 
+ *
+ * \author      Bjørn Bugge Grathwohl
+ * \date        February 2017
+ *
+ * \par Copyright 1995-2017 Shark Development Team
+ * 
+ * <BR><HR>
+ * This file is part of Shark.
+ * <http://shark-ml.org/>
+ * 
+ * Shark is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published 
+ * by the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ * 
+ * Shark is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ * 
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with Shark.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+//===========================================================================
+
+#ifndef SHARK_ALGORITHMS_DIRECT_SEARCH_OPERATORS_GRID
```

Comment:
In general: be a bit careful with generic names in the shark namespace. "closestIndices" is quite generic and can occur in different contexts meaning different things. Maybe encapsulate everything here into a class to get rid of this problem